### PR TITLE
chore(mdns): remove deprecated items

### PR DESCRIPTION
**Pull request by umgefahren**
## Description

Related: https://github.com/libp2p/rust-libp2p/issues/3647.

## Notes & open questions

Should be merged along with next breaking change to libp2p_mdns.

## Change checklist

- [x] I have performed a self-review of my own code
- [x] A changelog entry has been made in the appropriate crates

### CODE REVIEW

File: protocols/mdns/CHANGELOG.md
Comment (thomaseizinger) [inline protocols/mdns/CHANGELOG.md:8]: Should go under a `0.44.0` heading.

Comment (umgefahren) [reply protocols/mdns/CHANGELOG.md:8]: Done, I need get back into the mode of working on this repo...

**Comment by thomaseizinger** [comment]
We've now opened the merge window for breaking changes so this is good to go in once the merge conflicts are resolved.

**Review by thomaseizinger — APPROVED** [review]
Thanks!

**Pull request merged**
